src/execution_plan/ops/op_update.c

```diff
@@ -244,15 +243,17 @@ static void _EvalEntityUpdates(EntityUpdateCtx *ctx, GraphContext *gc,
 	// If the entity is a node, set its label if possible.
 	if(type == GETYPE_NODE) {
 		Node *n = (Node *)entity;
-		label_id = Graph_GetNodeLabel(gc->g, ENTITY_GET_ID(entity));
+		if(n->labelID != GRAPH_NO_LABEL) {
+			label_id = n->labelID;
+			label = n->label;
```

Comment:
might be a bit more compact
```
label_id =  n->labelID;
label = n->label;

if(n->labelID == GRAPH_NO_LABEL) {
...
}

Also please make sure to set
```
n->label = label;
n->labelID = label_id;`
``
at the end (if they were missing, this is the entire purpose of this PR_